File: programmers/level 2/phonenumber_solution.py
def solution(phone_book):
    phone_book = sorted(phone_book)
    answer = True
    for i in range(len(phone_book)):
        for j in range(i+1, len(phone_book)):
            if(phone_book[i] == phone_book[j][:len(phone_book[i])]):
                answer = False
                return answer
    return answer

File: programmers/level 2/phonenumber_solution.java
import java.util.HashMap;

class Solution {
    public boolean solution(String[] phone_book) {
        boolean answer = true;
        HashMap<String,String> map = new HashMap<>();//new에서 타입 파라미터 생략가능
        
        for(int i = 0; i < phone_book.length; i++) {
            map.put(phone_book[i], phone_book[i]);
        }
        
        for(int i = 0; i < phone_book.length; i++) {
            String temp = "";
            for(int j = 0; j < phone_book[i].length(); j++) {
                temp += phone_book[i].charAt(j);
                if(map.get(temp) != null && !map.get(temp).equals(phone_book[i])) {
                    return false;
                }
            }
        }
        
        return answer;
    }
}
